lib/types.ts

```diff
@@ -204,11 +212,23 @@ export type DeepOmit<T extends DeepOmitModify<Filter>, Filter> = T extends Built
   : T extends Promise<infer ItemType>
   ? ItemType extends DeepOmitModify<Filter>
     ? Promise<DeepOmit<ItemType, Filter>>
-    : T
-  : { [K in Exclude<keyof T, keyof Filter>]: T[K] } &
+    : T // explicitly mentioning optional properties, to work around TS making them required
+  : // see https://github.com/krzkaczor/ts-essentials/issues/118
```

Comment:
we can move this comment to the top of `DeepOmit` if prettier has issues with it